src/spikeinterface/core/recording_tools.py

```diff
@@ -68,12 +68,12 @@ def _init_binary_worker(recording, file_path_dict, dtype, byte_offest, cast_unsi
 
 
 def write_binary_recording(
-    recording,
-    file_paths,
-    dtype=None,
-    add_file_extension=True,
-    byte_offset=0,
-    auto_cast_uint=True,
+    recording: "BaseRecording",
+    file_paths: list[Path | str] | Path | str,
+    dtype: np.ndtype = None,
```

Comment:
I'm curious where did you find this type hint? dtype is one of the hard ones because I thought the official type was 

`numpy.typing.`DTypeLike` see docs [here](https://numpy.org/doc/stable/reference/typing.html#numpy.typing.DTypeLike).

But I think that degree of typing is too extreme, so in Neo we settle on `np.dtype` which is actually the construct to make a dtype object of whatever type you need. But maybe you've got some documentation of numpy typing that I could read?

Reply:
Basically went through the same reasoning as you do there.  I am using the dtype as a non-formal hint to the code user (this would not work with a static analyser I think) Plus, the official numpy dtype fails with some of the tests in neuroconv because it does not work with the inspect library in the standard library.

I am hoping that numpy dtype will stabilize as the python typing ecossytem matures and we are able to drop support of older python versions. Then we can switch comfortably, meanwhile this is better than nothing.


All of that reasoning was like a year or two ago and I have been operating on auto-pilot with that assumption. If things have changed and you find out there is a better way I am all ears.

Reply:
Works for me. I was hoping you had found something new :) 

But yeah I guess we just have to wait for things to stabilize. Thanks!